plugins/meta/portmap/chain.go

```diff
@@ -35,16 +35,17 @@ type chain struct {
 
 // setup idempotently creates the chain. It will not error if the chain exists.
 func (c *chain) setup(ipt *iptables.IPTables) error {
-	// create the chain
-	exists, err := chainExists(ipt, c.table, c.name)
-	if err != nil {
+
+	err := ipt.NewChain(c.table, c.name)
+	eerr, eok := err.(*iptables.Error)
```

Comment:
Type cast should be done when `err != nil`, right?

Reply:
We don't strictly need the check (ref https://play.golang.org/p/uSRCkdUHKni) but if you think it'd read nicer if I did, I can do that.

Reply:
I've updated these sections to break early on nil errors and get rid of the switch.